tests/test_roots.py

```diff
@@ -0,0 +1,81 @@
+from tests import SiteTest
+from pysite.constants import DISCORD_OAUTH_REDIRECT, DISCORD_OAUTH_AUTHORIZED
+
+class RootEndpoint(SiteTest):
+    """ Test cases for the root endpoint and error handling """
+
+    def test_index(self):
+        """ Check the root path responds with 200 OK """
+        response = self.client.get('/', 'http://pytest.local')
+        self.assertEqual(response.status_code, 200)
+
+    def test_info_index(self):
+        """ Check the info index path responds with a 301 """
+        response = self.client.get('/info')
+        self.assertEqual(response.status_code, 301)
+
+    def test_info_help(self):
+        """ Check the info help path responds with 200 OK """
+        response = self.client.get('/info/help')
+        self.assertEqual(response.status_code, 200)
+
+    def test_info_resources(self):
+        """ Check the info resources path responds with 200 OK """
+        response = self.client.get('/info/resources')
+        self.assertEqual(response.status_code, 200)
+
+    def test_info_resources_json(self):
+        """ Check the resources JSON loads correctly """
+        response = self.client.get('/static/resources.json')
+        self.assertEqual(response.status_code, 200)
+        self.assertIsInstance(response.json, dict)
+
+    def test_info_rules(self):
+        """ Check the info rules path responds with 200 OK """
+        response = self.client.get('/info/help')
+        self.assertEqual(response.status_code, 200)
+
+    def test_not_found(self):
+        """ Check paths without handlers returns 404 Not Found """
+        response = self.client.get('/nonexistentpath')
+        self.assertEqual(response.status_code, 404)
+
+    def test_error(self):
+        """ Check the /error/XYZ page """
+        response = self.client.get('/error/418')
+        self.assertEqual(response.status_code, 418)
```

Comment:
this only checks a single error page, i'm not sure if it's desired, but how about checking all of them with something like
```py
for code in HANDLED_ERROR_CODES:
    response = self.client.get(f'/error/{code}')
    self.assertEqual(response.status_code, code)
```

Reply:
looped done 👍